control-base/src/main/java/fi/nls/oskari/control/feature/AbstractWFSFeaturesHandler.java

```diff
@@ -44,13 +50,11 @@ public void init() {
                     ServiceFactory.getMapLayerService(),
                     ServiceFactory.getPermissionsService());
         };
-        if (myPlacesHelper == null) {
-            myPlacesHelper = new MyPlacesWFSHelper();
-        }
-        if (userlayerHelper == null) {
-            userlayerHelper = new UserLayerWFSHelper();
-        }
         this.wfsClient = new CachingWFSClient();
```

Comment:
_GeoJSON action route should not use CachingWFSclient so it doesn't fill out the cache as the bbox is always different_ 
First idea to pop into my head would be to create a WFSClient class with method tryGetFeatures(...) that the CachingWFSClient would then extend. Then we'd add a method `public WFSClient getWFSClient()` and call that here. The method would return a new CachingWFSClient by default and the GeoJSON action route could overload the method to return a non-caching version. Obviously this would also require changing the type of the field `wfsClient` from `CachingWFSClient` to `WFSClient`.